cpp/test/spatial/ann_ivf_flat.cu

```diff
@@ -206,25 +207,30 @@ class AnnIVFFlatTest : public ::testing::TestWithParam<AnnIvfFlatInputs> {
 
         index_params.add_data_on_build        = false;
         index_params.kmeans_trainset_fraction = 0.5;
-        auto index =
-          ivf_flat::build(handle_, index_params, database.data(), int64_t(ps.num_db_vecs), ps.dim);
 
-        rmm::device_uvector<int64_t> vector_indices(ps.num_db_vecs, stream_);
+        auto database_view = raft::make_device_matrix_view<const DataT>(
+          (const DataT*)database.data(), ps.num_db_vecs, ps.dim);
+
+        auto index = ivf_flat::build_index<DataT, IdxT>(handle_, database_view, index_params);
+
+        rmm::device_uvector<IdxT> vector_indices(ps.num_db_vecs, stream_);
         thrust::sequence(handle_.get_thrust_policy(),
                          thrust::device_pointer_cast(vector_indices.data()),
                          thrust::device_pointer_cast(vector_indices.data() + ps.num_db_vecs));
         handle_.sync_stream(stream_);
 
         int64_t half_of_data = ps.num_db_vecs / 2;
 
-        auto index_2 =
-          ivf_flat::extend<DataT, int64_t>(handle_, index, database.data(), nullptr, half_of_data);
+        auto half_of_data_view = raft::make_device_matrix_view<const DataT>(
+          (const DataT*)database.data(), static_cast<int>(half_of_data), ps.dim);
```

Comment:
Would you consider using `const_cast` here instead of a C-style cast?

Reply:
Is the cast even needed? I think it's clean enough to just express in the templates `raft::make_device_matrix_view<const DataT, int>(database.data(), half_of_data, ps.dim);`

Reply:
It's test code so I honestly wasn't particularly worried about it. I'll update it, though.